fem/kfespace.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+
+#ifndef MFEM_KFESPACE_HPP
+#define MFEM_KFESPACE_HPP
+
+#include "../config/config.hpp"
+
+#include "../general/karray.hpp"
+
+// *****************************************************************************
+MFEM_NAMESPACE
+
+// ***************************************************************************
+// * kFiniteElementSpace
+//  **************************************************************************
+class kFiniteElementSpace : public FiniteElementSpace
```

Comment:
I'm not sure if we should make k* versions of all the classes with kernels, or just back the kernels into the * classes themselves.  This may be worth discussing further.